models/users_sessions.sql

```diff
@@ -0,0 +1,77 @@
+--models/users_stitched.sql
+
+with stitched_visitor_ids as
+(
+  select
+    pageviews.customer_id
+    ,min(pageviews.visitor_id) visitor_id_stitched
+  from {{source('web_tracking', 'pageviews')}} as pageviews
+  group by 1
+)
+,users_stitched as
+(
+  select 
+    pageviews.id
+    ,stitched_visitor_ids.visitor_id_stitched as visitor_id
+    ,pageviews.device_type
+    ,pageviews.timestamp
+    ,pageviews.page
+    ,pageviews.customer_id
+  from {{source('web_tracking', 'pageviews')}} as pageviews
+  left join stitched_visitor_ids
+    on pageviews.customer_id = stitched_visitor_ids.customer_id
+  where pageviews.customer_id is not null
+  order by pageviews.customer_id
+)
```

Comment:
This does it 👍  Like the `min(pageviews.visitor_id)` to get the visitor id, that's clever! One thought... there may be cases where we've seen a `customer_id` associated with a `visitor_id` in the past, but then in subsequent page views we see the same `visitor_id` but logged out, so no `customer_id`, but we know (to an acceptable degree of certainty imo) who that visitor is from previous pageviews. I opted to id every customer_id <> visitor_id pairing then join on the visitor_id to tie missing visitors back to the customer. We may misidentify users where 2 different customers are on the same device/browser and share a visitor_id, but seems unlikely. Any thoughts on that?